include/mega/filesystem.h

```diff
@@ -362,8 +436,8 @@ struct MEGA_API FileSystemAccess : public EventTrigger
     void unescapefsincompatible(string*,FileSystemType) const;
```

Comment:
Yes, that is exactly the sort of issue that LocalPath is meant to help with.  In this case there are a couple of things - getlocalfstype was merged from a separate branch while LocalPath was still in development on another branch, so it needs use to be very aware of what the string content we are passing it is.   And also, LocalPath depends on FileSystemAccess, and if we add a dependency in the other direction the system becomes a bit more complex, and it would be more lines changed and needing review, so I thought on balance I would not change it at this point.   Perhaps in future we should change that, but I thought that the volume of changes would be the biggest problem with this PR.   If that part is working ok then it's less risky to convert it at this point too.

Reply:
Sure, let's not overcomplicate this PR. I thought I mention that, because it sounded to me there could be a bug here (coming from elswhere, not this PR). I've contacted Javier Gonzalez, he's taking a look on that.